### DISCUSSION TITLE
Is it possible to use seg images directly for generation?

### DISCUSSION THREAD

**Discussion by TheLearnerL** [Mikubill/sd-webui-controlnet#403] [Q&A]
I wanted the spatial structure of the environment to be more under my control, so I drew a seg image of my own.
![lQLPJwgK8buwTujNAnXNBViw7C9d5GzhzHcD9b45scALAA_1368_629 png_720x720q90g 拷贝](https://user-images.githubusercontent.com/32671428/221525757-25165d48-0b13-4fb4-ab11-d26d0359d513.png)
But after the seg pre-processing, the picture will be worse.
![image](https://user-images.githubusercontent.com/32671428/221525836-b06df262-8f9f-4e1e-89c4-c0037b771f05.png)
Is there a way to get contronet to use my original seg image for processing?

**Comment by toyxyz** [comment]
Just set the preprocessor to None. then it will work
![image](https://user-images.githubusercontent.com/8006000/221546471-88369b65-df85-43b0-95a0-a4feeb5ca67f.png)
